.argo-ci/ci.yaml

```diff
@@ -0,0 +1,158 @@
+    - name: app-revision
+      value: master
+    - name: app-org
+      value: argoproj
+    - name: app-repo
+      value: argo-cd-hello-world-app
+    - name: config-revision
+      value: master
+    - name: config-org
+      value: argoproj
+    - name: config-repo
+      value: argo-cd-hello-world-config
+    - name: argo-cd-sync-timeout
+      value: 600
+  volumes:
+  - name: docker-cred
+    secret:
+      secretName: docker-cred
+  - name: git-ssh-key
+    secret:
+      secretName: git-ssh-key
+      defaultMode: 256
+  templates:
+  - name: argo-cd-hello-world-ci
+    steps:
+    - - name: build
+        template: ci-dind
+        arguments:
+          parameters:
+          - name: cmd
+            value: make docker-build publish && git rev-parse HEAD | tee /tmp/git_sha.txt
+    - - name: update-preprod-image
+        template: ksonnet
+        arguments:
+          parameters:
+          - name: env
+            value: preprod
+          - name: cmd
+            value: "{{item}}"
+          artifacts:
+          - name: git-sha
+            from: "{{steps.build.outputs.artifacts.git-sha}}"
+        withItems:
+        - ./update-image.sh $ENV $ARGO_CD_APP_PREFIX $IMAGE_NAME $CONFIG_ORG $CONFIG_REPO $(cat /tmp/git_sha.txt)
+    - - name: deploy-preprod-env
+        template: argo-cd-deploy
+        arguments:
+          parameters:
+          - name: env
+            value: preprod
+          - name: cmd
+            value: "{{item}}"
+        withItems:
+        - ./argocd --auth-token $ARGO_CD_TOKEN --server $ARGO_CD_SERVER app sync $ARGO_CD_APP_PREFIX-$ENV && ./argocd --auth-token $ARGO_CD_TOKEN --server $ARGO_CD_SERVER app wait $ARGO_CD_APP_PREFIX-$ENV --timeout $ARGO_CD_SYNC_TIMEOUT
+
+  - name: argo-cd-deploy
+    inputs:
+      parameters:
+        - name: env
+        - name: cmd  
+    container:
+      image: argoproj/argocd-cli:v0.7.1
+      command: [sh, -c]
+      args: ["{{inputs.parameters.cmd}}"]
+      env:
+      - name: ENV
+        value: "{{inputs.parameters.env}}"
+      - name: ARGO_CD_APP_PREFIX
+        value: "{{workflow.parameters.argo-cd-app-prefix}}"
+      - name: ARGO_CD_SERVER
+        value: "{{workflow.parameters.argo-cd-server}}"
+      - name: ARGO_CD_TOKEN
+        valueFrom:
+          secretKeyRef:
+            name: argo-cd-token
+            key: auth-token
+      - name: ARGO_CD_SYNC_TIMEOUT
+        value: "{{workflow.parameters.argo-cd-sync-timeout}}"
+  - name: ksonnet
+    inputs:
+      parameters:
+      - name: env
+      - name: cmd
+      artifacts:
+      - name: git-sha
+        path: "/tmp/git_sha.txt"
+      - name: code
+        path: "/go/src/github.com/{{workflow.parameters.config-org}}/{{workflow.parameters.config-repo}}"
+        git:
+          repo: "https://github.com/{{workflow.parameters.config-org}}/{{workflow.parameters.config-repo}}.git"
+          revision: "{{workflow.parameters.config-revision}}"
+    container:
+      image: argoproj/argo-cd-tools:latest
```

Comment:
In our next release, lets get rid of this container and just put the necessary binaries (ksonnet, kustomize, argocd, helm) inside the argocd-cli container. This will simplify the example pipeline which depend on these tools.

Reply:
👍